### ISSUE TITLE
no-multi-assign doe’s not work for object values. 

### ISSUE THREAD

**Issue by DerZyklop** [eslint/eslint#13988]
**Tell us about your environment**

https://eslint.org/demo

**What parser are you using?**

default

**Please show your full configuration:**

☑️  &nbsp;no-multi-assign

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

Paste this into the textarea/code-box-thing:

```js
var myVar1;
var myVar2;

// Next line gets marked with error "no-multi-assign".
myVar1 = myVar2 = 123;

var myVar3;
var myObject = {
  // Next line should get marked with error "no-multi-assign" but does not.
  someKey: myVar3 = 123, 
};
```

**What did you expect to happen?**

Getting two errors in my source code:
5:10 - Unexpected chained assignment. (no-multi-assign)
10:12 - Unexpected chained assignment. (no-multi-assign)

**What actually happened? Please include the actual, raw output from ESLint.**

Getting one error in my source code:
5:10 - Unexpected chained assignment. (no-multi-assign)

**Are you willing to submit a pull request to fix this bug?**

I don’t know how to write rules or how to implement this. Sorry.

**Comment by mdjermanovic** [comment]
Hi @DerZyklop, thanks for the issue!

The `no-multi-assign` rule aims to disallow chained `=`, which isn't the case in the example.

If you want to disallow assignments in that position, you can configure [no-restricted-syntax](https://eslint.org/docs/rules/no-restricted-syntax) rule with `Property > AssignmentExpression.value` selector:

```js
/* eslint no-restricted-syntax: ["error", "Property > AssignmentExpression.value"] */

var myObject = {  
  someKey: myVar3 = 123 // error
};
```

[Online Demo](https://eslint.org/demo#eyJ0ZXh0IjoiLyogZXNsaW50IG5vLXJlc3RyaWN0ZWQtc3ludGF4OiBbXCJlcnJvclwiLCBcIlByb3BlcnR5ID4gQXNzaWdubWVudEV4cHJlc3Npb24udmFsdWVcIl0gKi9cblxudmFyIG15T2JqZWN0ID0geyAgXG4gIHNvbWVLZXk6IG15VmFyMyA9IDEyMyAvLyBlcnJvclxufTsiLCJvcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYVZlcnNpb24iOjUsInNvdXJjZVR5cGUiOiJzY3JpcHQiLCJlY21hRmVhdHVyZXMiOnt9fSwicnVsZXMiOnt9LCJlbnYiOnt9fX0=)

**Comment by DerZyklop** [comment]
@mdjermanovic  Works for me! Thank you! 

But.. assuming that you can call

```
let foo {
  bar: 3 // <- THIS
}
```

a value-**assign**ment to a object property. Thus i still think its valid to assume that `no-multi-assign` covers this case. This is the reason why i now hesitate to close this issue. 

I would be interested what other developers think about it.

**Comment by btmills** [comment]
I agree that this is working as intended. @DerZyklop I'm not clear what you're trying to show with your last code example. It currently has a syntax error. Did you mean `let foo = { bar: 3 }`? Initializing object properties inside a variable declaration is such a common pattern I can't see us every adding a rule to forbid it. If you want to do that, the `no-restricted-syntax` rule is the way to go.
